gcs/src/test/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemConfigurationPropertyTest.java

```diff
@@ -0,0 +1,118 @@
+package com.google.cloud.hadoop.fs.gcs;
+import com.google.cloud.hadoop.gcsio.GoogleCloudStorageFileSystemOptions;
+import com.google.cloud.hadoop.util.EntriesCredentialConfiguration;
+import com.google.common.collect.ImmutableList;
+import org.apache.hadoop.conf.Configuration;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import java.util.Collection;
+
+import static com.google.cloud.hadoop.fs.gcs.GoogleHadoopFileSystemTestBase.loadConfig;
+import static com.google.common.truth.Truth.assertThat;
+import static org.junit.Assert.assertThrows;
+
+@RunWith(JUnit4.class)
+public class GoogleHadoopFileSystemConfigurationPropertyTest {
+
+    @Test
+    public void testPropertyCreation() {
+        GoogleHadoopFileSystemConfigurationProperty<Integer>
+                NEW_KEY_WITH_DEPRECATED_KEY = new GoogleHadoopFileSystemConfigurationProperty<>(
+                "actual.key", 0, "deprecated.key");
+        assertThat(NEW_KEY_WITH_DEPRECATED_KEY.getKey()).isEqualTo("actual.key");
+
+        GoogleHadoopFileSystemConfigurationProperty<Integer>
+                NEW_KEY_WITHOUT_DEPRECATED_KEY = new GoogleHadoopFileSystemConfigurationProperty<>(
+                "actual.key", 0,null);
+        assertThat(NEW_KEY_WITHOUT_DEPRECATED_KEY.getDefault()).isEqualTo(0);
+    }
+
+    @Test
+    public void testCollectionTypeProperty() {
```

Comment:
Test methods should be split to test single test case and express this in method name:
`getStringCollection_throwsExceptionOnNonCollectionPreoperty`

Reply:
Sure, and following this example another proxy tests may look like this testProxyProperties_throwsExceptionWhenMissingProxyAddress?

Reply:
Yes, generally test name should explain what it tests and what result is expects.